### ISSUE TITLE
Feature Request: Cloud Memorystore Resources

### ISSUE THREAD

**Issue by dimitropoulos** [terraform-providers/terraform-provider-google#1467]
Now in Beta: Cloud Memorystore fills an important void in the GCP landscape.  It would be fantastic to have support for it.

### References
- https://cloud.google.com/memorystore/
- https://cloud.google.com/sdk/gcloud/reference/beta/redis/
- https://github.com/GoogleCloudPlatform/google-cloud-go
- https://github.com/google/google-api-go-client

**Comment by mrparkers** [comment]
If nobody has signed up to take this on, I volunteer to work on this over the weekend.

**Comment by pdecat** [comment]
@mrparkers I'm not sure its worth to implement it by hand given new resources are now generated by the @modular-magician : https://github.com/modular-magician/terraform-provider-google / https://github.com/GoogleCloudPlatform/magic-modules

**Comment by mrparkers** [comment]
@pdecat Cool, I didn't know that existed.  How long does it usually take for resources to be implemented automatically?  I'd like for this resource to exist before the next release of this provider, so I really don't mind doing this manually if it means achieving that.

I was also looking for something to do this weekend so I may just do it anyways :smile: if nothing else, it's an opportunity to help me learn Go

**Comment by danawillow** [comment]
@mrparkers it depends on the API in question and whether magic modules has support for all the features (for lack of a better term) of that API. I was playing around with this the other day and found one that needs to be added: https://github.com/GoogleCloudPlatform/magic-modules/issues/182. I was planning on doing this early next week, and I _think_ that'll be the only thing that blocks this, so if you could hold off that would be great (since it's easier if we generate things from the onset than to generate something that's already been done by hand). I'll let you know if I change my mind.

If you're looking for things to work on though, there are plenty of other open issues 😉

**Comment by mrparkers** [comment]
@danawillow sounds good, I can look for something else.

This is a bit off-topic, but does this repository have a concept of tagging "good first issues"?  Although I have contributed before, I'm still very new to Go and I was looking for ways to help with my learning experience.  Usually new resources are pretty easy, which is why I wanted to work on this one.

**Comment by paddycarver** [comment]
So we've got the `good first issue` and `help wanted` labels, which are made for tracking things we _believe_ to be straightforward enough that someone could tackle them as a first contribution. We haven't gotten great at using them (or at least I haven't), so they may be of limited use. There's also `easy`, which is going to go away, because I dislike the implications of that one, but historically that has been how we tracked things.
